### PR TITLE
Add maintainer docs

### PR DESCRIPTION
**Pull request by randomvariable**
Signed-off-by: Naadir Jeewa <jeewan@vmware.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Documenting all the places that need updating for maintainer status.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,3 +13,35 @@ Kubernetes projects require that you sign a Contributor License Agreement (CLA)
...
+
+The complete list of tasks required to set up maintainer status
+follow:
+
+* Open PR to add Github username to the OWNERS_ALIASES file under cluster-api-aws-maintainers
+* Open PR to add Github username to cluster-api-provider-aws-admins and cluster-api-provider-aws-maintainers
+to https://github.com/kubernetes/org/blob/main/config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml
+* Open PR to add Github username to https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api-provider-aws/OWNERS
+* Open PR to add Github username to https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-cluster-api-aws/OWNERS
+* Open PR to add Github username to https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api-provider-aws/OWNERS
```
Comment (sedefsavas) [inline CONTRIBUTING.md:46]: Duplicate
```suggestion
```

**Comment by richardcase** [comment]
Apart from the duplicate line this LGTM

**Comment by randomvariable** [comment]
Done

**Pull request merged**